.github/workflows/github-actions.yml

```diff
@@ -1,8 +1,58 @@
 name: GitHub Actions AWS connection
 run-name: ${{ github.actor }} triggered the Github actions
-on: [push]
+
+on:
+  pull_request:
+    branches:
+      - main  # Trigger on PR to the 'main' branch (adjust to your target branch)
+  push:
+    branches: 
+      - main  # Runs on push to main branch only (for Explore-GitHub-Actions)
 
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'  # Ensure linting only runs for PR events
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+
+      - name: Set up Python (for linting tools like flake8, black, yamllint)
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'  # Adjust based on your Python version
+
+      - name: Install linting dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 black yamllint  # Install your linting tools
+
+      - name: Run Linting (Python files, YAML files)
+        run: |
+          ./test/lint.sh  # Run your custom lint script from the test folder
+
+  unit-test:
```

Comment:
Is this section about unit tests your code or Brandon's? Either way the title should probably reflect that this PR includes CI for unit tests as well.